### PR TITLE
Extract parse_signed_request in a Utils module for becoming callable fro...

### PR DESCRIPTION
**Pull request by Vrael**
Related to issue: mkdynamic/omniauth-facebook#177

- [x] Extract in a Utils module some private methods from Strategies::Facebook
- [x] rake test checked

### CODE REVIEW

**Comment by mkdynamic** [comment]
Very nice, thanks. Can you please move into it's own file, within the OmniAuth::Facebook namespace?

File: lib/omniauth/strategies/facebook.rb [hunk 1]
```diff
@@ -5,13 +5,42 @@
 require 'uri'
 
 module OmniAuth
+  module Utils
+    class UnknownSignatureAlgorithmError < NotImplementedError; end
+    
+    SUPPORTED_ALGORITHM = 'HMAC-SHA256'
+    
+    def Utils.parse_signed_request(value, secret)
```
Comment (mkdynamic) [inline lib/omniauth/strategies/facebook.rb]: It is more idiomatic to write these methods as `self. parse_signed_request`. etc.

**Comment by mkdynamic** [comment]
Another thought: it may be nicer to change the methods in `Util` to regular instance methods and then include the module into `OmniAuth::Strategies::Facebook`. Probably more in line with what @simi was thinking in https://github.com/mkdynamic/omniauth-facebook/issues/177#issuecomment-70416039.

**Comment by simi** [comment]
My eyes are here also. I'll rewiew this tomorrow.
Dne 19.1.2015 21:49 "Mark Dodwell" <notifications@github.com> napsal(a):

**Comment by Vrael** [comment]
@mkdynamic why do you say?: 
> it may be nicer to change the methods in Util to regular instance methods
>

I mean, when methods are in a instance, the class must be instanced before for using them, and in Ruby the modules are for methods which doesn't need be instantiated, for example functions in a library. So what advantage do we obtain to becoming them into instance methods?

**Comment by simi** [comment]
I think this should be nice syntax for this.

```ruby
signed_request = OmniAuth::Facebook::SignedRequest.new(value, secret)
signed_request.payload #=> parsed request
```

**Comment by simi** [comment]
Hello @Vrael. Are you still interested in finishing this? I can finish it also if you're out of time.

**Comment by Vrael** [comment]
@simi I'm a little busy these days. If you have time to finish it then perfect!

**Comment by simi** [comment]
merged in #183

**Pull request closed without merging**